### PR TITLE
Refactor serializers

### PR DESCRIPTION
**Pull request by JauntyJames**
Changed up serializers to return average rating, specific data for the show page, and user info for individual reviews

### CODE REVIEW

File: app/javascript/react/components/ReviewForm.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            checked={props.rating==='80'}
+            onChange={handleRatingChange}
+          />
+        </label>
+        <label>5
+          <input name="rating" type="radio" value="100"
+            checked={props.rating==='100'}
+            onChange={handleRatingChange}
+          />
+        </label>
```
Comment (bkincart) [inline app/javascript/react/components/ReviewForm.js]: This isn't all that DRY - could we factor this field out into its own `const` component, and hand in props (rating, value, etc)?

File: app/javascript/react/containers/ReviewFormContainer.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  ratingChange(value) {
+    this.setState({ rating: value })
+  }
+
+  handleSubmit() {
+    let formPayload = {
+      review: {
+        body: this.state.bodyText,
+        rating: this.state.rating,
+        user_id: 1,  // this is a temporary hardcorde for testing puroposes, change to: props.userId,
```
Comment (bkincart) [inline app/javascript/react/containers/ReviewFormContainer.js]: I wouldn't want to merge this in with the hardcoded value! Can we get this updated yet, or is there something you're waiting on?

File: app/javascript/react/containers/WizardShowContainer.js [hunk 3]
```diff
@@ -30,7 +33,49 @@ class WizardShowContainer extends Component {
...
+
+  addNewReview(formPayload) {
+
+    fetch('/api/v1/reviews', {
+      credentials: 'same-origin',
+      method: 'post',
+      body: JSON.stringify(formPayload),
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Requested-With': 'XMLHttpRequest',
```
Comment (bkincart) [inline app/javascript/react/containers/WizardShowContainer.js:45]: Can you explain why you chose to include this please?

**Review by bkincart — COMMENTED** [review]
These changes apply if the form stuff will be merged in  with this PR!

File: app/javascript/react/containers/WizardShowContainer.js (hunk 3 above)
Comment (iBrooks) [reply app/javascript/react/containers/WizardShowContainer.js:45]: This is part of passing the authenticity token to the api endpoint
added another header as well

File: app/javascript/react/components/ReviewForm.js (hunk 1 above)
Comment (nwalberts) [reply app/javascript/react/components/ReviewForm.js]: Still seeing this as well!

File: app/javascript/react/containers/ReviewFormContainer.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  reviewChange(value) {
+    this.setState({ bodyText: value })
+  }
+
+  render() {
+    let passDownReviewChange = (value) => { this.reviewChange(value) }
+    let passDownRatingChange = (value) => { this.ratingChange(value) }
+    let passDownSubmit = () => { this.handleSubmit() }
+
```
Comment (nwalberts) [inline app/javascript/react/containers/ReviewFormContainer.js]: you shouldnt need line 28, just pass down handleSubmit

File: app/javascript/react/components/ReviewForm.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  let handleBodyChange = (event) => {
+    let value = event.target.value
+    props.passDownReviewChange(value);
+  }
+
+  let handleRatingChange = (event) => {
+    let value = event.target.value
+    props.passDownRatingChange(value);
+  }
+
```
Comment (nwalberts) [inline app/javascript/react/components/ReviewForm.js]: this is okay, but would probably less confusing if you do the standard practice for handleChange methods and define them in the parent container

File: app/serializers/wizard_serializer.rb [hunk 6]
```diff
@@ -1,3 +1,6 @@
 class WizardSerializer < ActiveModel::Serializer
-  attributes :id, :name, :img_url, :description, :creator_id, :created_at
+  attributes :id, :name, :img_url, :created_at, :average_rating
+  # has_many :reviews
+
+
```
Comment (nwalberts) [inline app/serializers/wizard_serializer.rb]: Delete this commented out code

File: app/javascript/react/containers/WizardShowContainer.js (hunk 3 above)
Comment (nwalberts) [reply app/javascript/react/containers/WizardShowContainer.js:45]: See my above comment on the api reviews controller. Once you have made that change, then you can remove those two additional headers

File: app/controllers/api/v1/reviews_controller.rb [hunk 7]
```diff
@@ -1,6 +1,6 @@
 class Api::V1::ReviewsController < ApplicationController
-
```
Comment (nwalberts) [inline app/controllers/api/v1/reviews_controller.rb:2]: `skip_before_action :verify_authenticity_token`

Should be able to allow you to remove those headers

**Pull request merged**